kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -237,8 +274,14 @@ class KotlinJsonAdapterFactory : JsonAdapter.Factory {
 
       val name = jsonAnnotation?.name ?: property.name
       val resolvedPropertyType = resolve(type, rawType, property.returnType.javaType)
-      val adapter = moshi.adapter<Any>(
-          resolvedPropertyType, Util.jsonAnnotations(allAnnotations.toTypedArray()), property.name)
+      val adapter = if(failureStrategy == FailureStrategy.ACCUMULATE_ERRORS &&
+              !property.returnType.isMarkedNullable) {
```

Comment:
when `returnType.isMarkedNullable` is true the adapter is already `nullSafe()`

if there is a better way to do this let me know and I will update the PR.

Reply:
I'm not quite sure I understand (but I'm late to this PR so I might be missing prior context). Can you elaborate a bit more?